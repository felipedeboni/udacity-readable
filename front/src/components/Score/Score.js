import React from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import {
  faChevronUp,
  faChevronDown
} from '@fortawesome/fontawesome-free-solid';
import './Score.css';

const Score = ({ id, score, onVoteUp, onVoteDown }) => {
  return (
    <div className="score">
      <span className="score__number">{score}</span>
      <FontAwesomeIcon
        className="score__vote score__vote--up"
        icon={faChevronUp}
        onClick={() => onVoteUp(id)}
      />
      <FontAwesomeIcon
        className="score__vote score__vote--down"
        icon={faChevronDown}
        onClick={() => onVoteDown(id)}
      />
    </div>
  );
};

Score.propTypes = {
  id: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  onVoteUp: PropTypes.func.isRequired,
  onVoteDown: PropTypes.func.isRequired
};

export default Score;
