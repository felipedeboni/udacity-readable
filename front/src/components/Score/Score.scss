@keyframes slideUp {
    0% {
        transform: translateY(0);
    }
    50% {
        transform: translateY(-30%);
    }
    100% {
        transform: translateY(0);
    }
}

@keyframes slideDown {
    0% {
        transform: translateY(0);
    }
    50% {
        transform: translateY(30%);
    }
    100% {
        transform: translateY(0);
    }
}

.score {
  color: #999;
  font-size: 13px;

  &__number {
    font-weight: 600;
    margin-right: 5px;
  }

  &__vote {
    cursor: pointer;
    margin: 0 4px;
    transition: color .2s ease-in;

    &--up:hover {
      color: #3c763d;
      -webkit-animation: slideUp 0.5s linear;
      animation: slideUp 0.5s linear;
    }

    &--down:hover {
      color: #a94442;
      -webkit-animation: slideDown 0.5s linear;
      animation: slideDown 0.5s linear;
    }

    &:hover {
      -webkit-animation-iteration-count: infinite;
      animation-iteration-count: infinite;
      -webkit-animation-fill-mode: forwards;
      animation-fill-mode: forwards
    }
  }
}

@media (min-width: 767px) {
  .score {
    font-size: 16px;

    &__number {
      margin-right: 2px;
    }

    &__vote {
      margin-left: 1px;
      margin-right: 1px;
    }
  }
}
