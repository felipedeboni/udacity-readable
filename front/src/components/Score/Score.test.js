import React from 'react';
import Score from './Score';
import { shallow, mount } from 'enzyme';

describe('Score', () => {
  it('renders without crashing', () => {
    const onVoteUp = jest.fn();
    const onVoteDown = jest.fn();
    const test = (
      <Score id="1" score={1} onVoteUp={onVoteUp} onVoteDown={onVoteDown} />
    );

    expect(shallow(test)).toMatchSnapshot();
  });

  it('calls vote up', () => {
    const onVoteUp = jest.fn();
    const onVoteDown = jest.fn();
    const test = mount(
      <Score id="1" score={1} onVoteUp={onVoteUp} onVoteDown={onVoteDown} />
    );
    test
      .find('.score__vote--up')
      .first()
      .simulate('click');
    expect(onVoteUp).toHaveBeenCalled();
  });

  it('calls vote down', () => {
    const onVoteUp = jest.fn();
    const onVoteDown = jest.fn();
    const test = mount(
      <Score id="1" score={1} onVoteUp={onVoteUp} onVoteDown={onVoteDown} />
    );
    test
      .find('.score__vote--down')
      .first()
      .simulate('click');
    expect(onVoteDown).toHaveBeenCalled();
  });
});
