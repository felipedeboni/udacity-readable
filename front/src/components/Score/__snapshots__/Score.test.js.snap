// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders without crashing 1`] = `
<div
  className="score"
>
  <span
    className="score__number"
  >
    1
  </span>
  <FontAwesomeIcon$1
    border={false}
    className="score__vote score__vote--up"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f077",
          "M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z",
        ],
        "iconName": "chevron-up",
        "prefix": "fas",
      }
    }
    listItem={false}
    mask={null}
    name=""
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    transform={null}
  />
  <FontAwesomeIcon$1
    border={false}
    className="score__vote score__vote--down"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f078",
          "M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z",
        ],
        "iconName": "chevron-down",
        "prefix": "fas",
      }
    }
    listItem={false}
    mask={null}
    name=""
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    transform={null}
  />
</div>
`;
